src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/ApplicationRepository.cs

```diff
@@ -118,8 +118,10 @@ public async Task<Application> Create(Application item)
         }
 
         /// <inheritdoc/>
-        public async Task<Application> Update(Application item)
+        public async Task<Application> Update(Application cachedApplication)
         {
+            Application item = DeepClone(cachedApplication);
```

Comment:
What is the theory for this fix?

Reply:
The PreProcess logic changes the application id before the changes are being saved to CosmosDB. The problem is that the object being changed is the same exact object (memory reference) as the application instance stored in the MemoryCache, and used in operations like the Instance Post. 

There are better fixes, but this was quick and easy. We later decided against pushing it out on a Friday evening. We can take a second look on Monday.